pkg/sql/comment_on_column_test.go

```diff
@@ -109,118 +105,130 @@ func TestCommentOnColumnTransaction(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
-	params, _ := tests.CreateTestServerParams()
-	s, db, _ := serverutils.StartServer(t, params)
-	defer s.Stopper().Stop(context.Background())
-
-	if _, err := db.Exec(`
+	runCommentOnTests(t, func(db *gosql.DB) {
+		if _, err := db.Exec(`
 		CREATE DATABASE d;
 		SET DATABASE = d;
 		CREATE TABLE t (c INT);
 	`); err != nil {
-		t.Fatal(err)
-	}
-
-	if _, err := db.Exec(`COMMENT ON COLUMN t.c IS 'foo'`); err != nil {
-		t.Fatal(err)
-	}
+			t.Fatal(err)
+		}
 
-	if _, err := db.Exec(`ALTER TABLE t DROP COLUMN c`); err != nil {
-		t.Fatal(err)
-	}
+		if _, err := db.Exec(`COMMENT ON COLUMN t.c IS 'foo'`); err != nil {
+			t.Fatal(err)
+		}
 
-	row := db.QueryRow(`SELECT comment FROM system.comments LIMIT 1`)
-	var comment string
-	err := row.Scan(&comment)
-	if !errors.Is(err, gosql.ErrNoRows) {
-		if err != nil {
+		if _, err := db.Exec(`ALTER TABLE t DROP COLUMN c`); err != nil {
 			t.Fatal(err)
 		}
 
-		t.Fatal("comment remaining in system.comments despite drop")
-	}
+		row := db.QueryRow(`SELECT comment FROM system.comments LIMIT 1`)
+		var comment string
+		err := row.Scan(&comment)
+		if !errors.Is(err, gosql.ErrNoRows) {
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			t.Fatal("comment remaining in system.comments despite drop")
+		}
+	})
 }
 
 func TestCommentOnAlteredColumn(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	params, _ := tests.CreateTestServerParams()
-	s, db, _ := serverutils.StartServer(t, params)
-	defer s.Stopper().Stop(context.Background())
-	expectedComment := "expected comment"
+	runCommentOnTests(t, func(db *gosql.DB) {
+		expectedComment := "expected comment"
 
-	if _, err := db.Exec(`
+		if _, err := db.Exec(`
 		CREATE DATABASE d;
 		SET DATABASE = d;
 		SET enable_experimental_alter_column_type_general = true;
 		CREATE TABLE t (c INT);
 	`); err != nil {
-		t.Fatal(err)
-	}
+			t.Fatal(err)
+		}
 
-	if _, err := db.Exec(`COMMENT ON COLUMN t.c IS 'first comment'`); err != nil {
-		t.Fatal(err)
-	}
+		if _, err := db.Exec(`COMMENT ON COLUMN t.c IS 'first comment'`); err != nil {
+			t.Fatal(err)
+		}
 
-	if _, err := db.Exec(`ALTER TABLE t ALTER COLUMN c TYPE character varying;`); err != nil {
-		t.Fatal(err)
-	}
-	if _, err := db.Exec(
-		fmt.Sprintf(`COMMENT ON COLUMN t.c IS '%s'`, expectedComment)); err != nil {
-		t.Fatal(err)
-	}
-	row := db.QueryRow(`SELECT comment FROM system.comments LIMIT 1`)
-	var comment string
-	if err := row.Scan(&comment); err != nil {
-		t.Fatal(err)
-	}
+		if _, err := db.Exec(`ALTER TABLE t ALTER COLUMN c TYPE character varying;`); err != nil {
+			t.Fatal(err)
+		}
+		if _, err := db.Exec(
+			fmt.Sprintf(`COMMENT ON COLUMN t.c IS '%s'`, expectedComment)); err != nil {
+			t.Fatal(err)
+		}
+		row := db.QueryRow(`SELECT comment FROM system.comments LIMIT 1`)
+		var comment string
+		if err := row.Scan(&comment); err != nil {
+			t.Fatal(err)
+		}
+
+		if expectedComment != comment {
```

Comment:
This logic repeats itself, would it make sense to just have runCommentOnTests just be fully structured, so that it takes structures from the tests and runs the logic? Or is something not shared here that I'm missing

Reply:
Yeah, good point. There are a few tests that have the same pattern as this test. But still there're tests that don't share this structure. I'll leave it as is.